parkings/utils.py

```diff
@@ -0,0 +1,18 @@
+from django.db import transaction
+
+from parkings.models import Parking
+from parkings.models.parking import AbstractParking, ArchivedParking
+
+
+def archive_parkings(time):
+    parkings_to_archive = Parking.objects.filter(time_end__lte=time)
+
+    for parking in parkings_to_archive:
+        with transaction.atomic():
+            fields = [field.name for field in AbstractParking._meta.fields]
+            values = dict([(field, getattr(parking, field)) for field in fields])
+            archived_parking = ArchivedParking(**values)
+            archived_parking.save()
+            parking.delete()
```

Comment:
I think this would be better as manager method in the ParkingsQuerySet.

I think the methods in the ParkingQuerySet could be something like this:
```python

# in the management command
    def handle(...)
        parkings_to_archive = Parking.objects.ends_before(before)
        if confirm:
             print(parkings_to_archive.count()) # etc.
        parkings_to_archive.archive()


class ParkingQuerySet(...):
    def ends_before(self, time):
        self.exclude(time_end=None).filter(time_end__lt=time)

    def archive(self):
        for parking in self:
            parking.archive()

class Parking(...):
    def archive(self):
        archived_parking = self.make_archived_parking()
        with transaction.atomic():
            archived_parking.save()
            self.delete()
            return archived_parking

    def make_archived_parking(self):
        fields = [x.name for x in self._meta.fields]
        values = {x: getattr(self, x) for x in fields}
        return ArchivedParking(**values)


class ArchivedParking(...):
    def archive(self):
        return self
```